packages/js/product-editor/src/components/header/woo-more-menu-item/woo-more-menu-item.tsx

```diff
@@ -1,37 +1,32 @@
 /**
  * External dependencies
  */
-import { isEmpty } from 'lodash';
+import { Slot, Fill } from '@wordpress/components';
+import { createElement } from '@wordpress/element';
 import {
-	createSlotFill,
-	Slot as BaseSlot,
-	Fill as BaseFill,
-} from '@wordpress/components';
-import { createElement, Fragment } from '@wordpress/element';
+	createOrderedChildren,
+	sortFillsByOrder,
+} from '@woocommerce/components';
 
-type WooProductMoreMenuSlot = React.FC< BaseSlot.Props >;
+export const WC_PRODUCT_MORE_MENU_SLOT_NAME = 'WooProductMenuMenuItem';
 
-type WooProductMoreMenuFill = React.FC< BaseFill.Props > & {
-	Slot?: WooProductMoreMenuSlot;
+export const WooProductMoreMenuItem: React.FC< {
+	children?: React.ReactNode;
+	order?: number;
+} > & {
+	Slot: React.FC< Slot.Props >;
+} = ( { children, order = 1 } ) => {
+	return (
+		<Fill name={ WC_PRODUCT_MORE_MENU_SLOT_NAME }>
+			{ ( fillProps: Fill.Props ) => {
+				return createOrderedChildren( children, order, fillProps );
```

Comment:
I'm a little confused with this function name. At first, I thought the children were being ordered twice (both in `sortFillsByOrder` and here), but I see that's not the case. I wonder if this could be renamed.

Reply:
I agree it is confusing. I also think we could create a reusable `Fill` that encapsulated all of this so we didn't have to repeat this pattern everywhere. I'll create an issue to consider improving our ordered slot fill technique.